tests/unit/src/main/scala/tests/BaseWorksheetLspSuite.scala

```diff
@@ -5,323 +5,335 @@ import scala.meta.internal.metals.UserConfiguration
-             |  }
-             |}
-             |/a/src/main/scala/foo/Main.worksheet.sc
-             |identity(42)
-             |val name = sourcecode.Name.generate.value
-             |""".stripMargin
-        )
-        _ <- server.didOpen("a/src/main/scala/foo/Main.worksheet.sc")
-        _ <- server.didSave("a/src/main/scala/foo/Main.worksheet.sc")(identity)
-        identity <- server.completion(
-          "a/src/main/scala/foo/Main.worksheet.sc",
-          "identity@@"
-        )
-        _ = assertNoDiff(identity, "identity[A](x: A): A")
-        generate <- server.completion(
-          "a/src/main/scala/foo/Main.worksheet.sc",
-          "generate@@"
-        )
-        _ = assertNoDiff(generate, "generate: Name")
-        _ = assertNoDiagnostics()
-        _ = assertNoDiff(
-          client.workspaceDecorations,
-          """|identity(42) // 42
-             |val name = sourcecode.Name.generate.value // "name"
-             |""".stripMargin
-        )
-      } yield ()
-    }
 
-  testAsync("render") {
-    for {
-      _ <- server.initialize(
-        s"""
-           |/metals.json
-           |{"a": {"scalaVersion": "$scalaVersion"}}
-           |/a/src/main/scala/Main.worksheet.sc
-           |import java.nio.file.Files
-           |val name = "Susan"
-           |val greeting = s"Hello $$name"
-           |println(greeting + "\\nHow are you?")
-           |1.to(10).toVector
-           |val List(a, b) = List(42, 10)
-           |""".stripMargin
-      )
-      _ <- server.didOpen("a/src/main/scala/Main.worksheet.sc")
-      _ = assertNoDiagnostics()
-      _ = assertNoDiff(
-        client.workspaceDecorations,
-        """|
-           |import java.nio.file.Files
-           |val name = "Susan" // "Susan"
-           |val greeting = s"Hello $name" // "Hello Susan"
-           |println(greeting + "\nHow are you?") // Hello Susan
-           |1.to(10).toVector // Vector(1, 2, 3, 4, 5, 6, 7, 8, 9, 10)
-           |val List(a, b) = List(42, 10) // a=42, b=10
-           |""".stripMargin
-      )
-      _ = assertNoDiff(
-        client.workspaceDecorationHoverMessage,
-        """|import java.nio.file.Files
-           |val name = "Susan"
-           |name: String = "Susan"
-           |val greeting = s"Hello $name"
-           |greeting: String = "Hello Susan"
-           |println(greeting + "\nHow are you?")
-           |// Hello Susan
-           |// How are you?
-           |1.to(10).toVector
-           |res1: Vector[Int] = Vector(1, 2, 3, 4, 5, 6, 7, 8, 9, 10)
-           |val List(a, b) = List(42, 10)
-           |a: Int = 42
-           |b: Int = 10
-           |""".stripMargin
-      )
-    } yield ()
-  }
+  scalaVersions.foreach((scalaVersion: String) =>
+    ScalaVersion
+      .fromString(scalaVersion)
+      .fold(
+        (e: ScalaVersion.ScalaVersionParseError) => fail(e.errorMessage), {
+          (sv: ScalaVersion) =>
+            if (ScalaVersion.validScalaVersionForCurrentJDK(sv)) {
```

Comment:
Would be nice to avoid adding a custom “ScalaVersion” class with its own parsing logic just for this condition. Could we solve this problem instead with a hardcoded set of Scala version strings that don’t work in Java 11? 🤔

Reply:
We can do what you suggest, or we can use more simple parsing logic. It is my opinion both are a bit more brittle and difficult to debug in the case of typos in the input. 

I'll throw up two other branches so we can discuss, but I'll defer to what ever you'd like.